### PR TITLE
Simplify rustbuild Rust beta downloading.

### PR DESCRIPTION
**Pull request by frewsxcv**
No need to shell out to curl or WebClient when we can just use the
urllib2 library built
 into Python.

### CODE REVIEW

**Comment by xen0n** [comment]
No the `urllib2` module is Python 2-only, considering #36509 just landed a week ago I think we should only use Py3k-compatible modules from now on.

This should be enough without introducing `six` as an additional dep:

```python
try:
    from urllib2 import urlopen
except ImportError:
    from urllib.request import urlopen
```

File: src/bootstrap/bootstrap.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import sys
 import tarfile
 import tempfile
+import urllib2
```
Comment (xen0n) [inline src/bootstrap/bootstrap.py]: No the `urllib2` module is Python 2-only, considering #36509 just landed a week ago I think we should only use Py3k-compatible modules from now on.

This should be enough without introducing `six` as an additional dep:

```python
try:
    from urllib2 import urlopen
except ImportError:
    from urllib.request import urlopen
```

Comment (frewsxcv) [inline src/bootstrap/bootstrap.py]: Ah, didn't realize we supported Python 3, great! Will make the change.

**Comment by sanxiyn** [comment]
Note that pre-2.7.9 Python does not verify SSL certificates, but curl and WebClient do. Maybe we should warn on old versions of Python?

**Comment by alexcrichton** [comment]
Thanks for the PR! This was originally done in Cargo as well (where much of this is from) but [was reverted](https://github.com/rust-lang/cargo/pull/1347) for a few reasons:

* urlllib doesn't have proxy support, but curl does
* Python 2/3 compat (looks like that's handled here though)
* Spotty support for https in python, including SNI and such

Since this is intended to be "the most robust method of downloading we can find" it may not be super necessary to unify everywhere (it seems to work well today). For me the lack of proxy plus the spotty support for https in python is a bit of a deal breaker :(

**Comment by frewsxcv** [comment]
Yeah, thought about @sanxiyn's comment this morning and agree this is probably not a desired change. From what I've heard, Python `2.7.9 > n >= 2.7.0` (very unfortunately) still gets used quite a bit in industry, which results in security issues when using urllib2.

**Pull request closed without merging**